src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -115,6 +125,32 @@ protected Endpoint getEndpoint() {
         return Endpoint.INTERNALUSERS;
     }
 
+    @Override
+    protected void handleGet(final RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException {
+        final String resourcename = request.param("name");
+
+        SecurityDynamicConfiguration<?> configuration = load(getConfigName(), true);
+        filter(configuration);
+
+        String filterBy = request.param("filterBy", "all");
+
+        if (filterBy.equalsIgnoreCase("internal") || filterBy.equalsIgnoreCase("service")) {
```

Comment:
Move these types of filter into an enum, these are public API members after this code ships so we need to be sure they won't break.